sacred/experiment.py

```diff
@@ -483,9 +486,21 @@ def get_default_options(self) -> dict:
         its default value.
 
         """
-        _, _, internal_usage = self.get_usage()
-        args = docopt(internal_usage, [])
-        return {k: v for k, v in args.items() if k.startswith("--")}
+        default_options = {}
+        for option in self.all_cli_options:
```

Comment:
It's not clear what is the intention of this part of the code. Couldn't this be under a function with an appropriate naming ? 
Also, The False/None confuse me. Could it be only boolean ?

Reply:
>It's not clear what is the intention of this part of the code. 

This creates a dictionary with the flags as key and the default value of the command line option as values. There is some duplication due to backward compatibility purposes.

> Couldn't this be under a function with an appropriate naming ?

It already is. The function is named `get_default_options` which is very explicit.

> Also, The False/None confuse me. Could it be only boolean ?

It wouldn't be semantically correct (also note that I didn't decide this behavior, it's already the case since a long time ago). Some option are flags, so booleans make sense, some options expect a string argument given from the command line. So in this case using booleans as default values is just confusing. `None` is what is semantically correct in python when the value is not present.

Reply:
The default options should be possibly in the responsibility of the command line option. However, let's address this in a separate PR.